### PR TITLE
fix(vector): vector PodMonitor interval configuration

### PR DESCRIPTION
**Pull request by savanipoojan78**
Very simple fix, the template for vector podMonitor didn't have any interval configuration

### CODE REVIEW

File: charts/vector/values.yaml [hunk 1]
```diff
@@ -380,6 +380,10 @@ podMonitor:
   port: prom-exporter
   # podMonitor.path -- Override the path to scrape.
   path: /metrics
+  # podMonitor.interval -- Override the interval to scrape.
+  interval: 60s
+  # podMonitor.scrapeTimeout -- Override the scrapeTimeout to scrape.
+  # scrapeTimeout: 30s
```
Comment (neuronull) [inline charts/vector/values.yaml:386]:
```suggestion
  # podMonitor.interval -- Override the interval at which metrics should be scraped.
  interval:
  # podMonitor.scrapeTimeout -- Override the timeout after which the scrape is ended.
  scrapeTimeout:
```

File: charts/vector/templates/podmonitor.yaml [hunk 2]
```diff
@@ -19,6 +19,10 @@ spec:
   podMetricsEndpoints:
     - port: {{ .Values.podMonitor.port }}
       path: {{ .Values.podMonitor.path }}
+      interval: {{ .Values.podMonitor.interval }}
```
Comment (neuronull) [inline charts/vector/templates/podmonitor.yaml:22]:
```suggestion
      {{- if .Values.podMonitor.interval }}
      interval: {{ .Values.podMonitor.interval }}
      {{- end }}
```

**Review by neuronull — COMMENTED** [review]
👋  hi @savanipoojan78 , thanks for the contribution!

I made some suggestions based on https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.PodMetricsEndpoint

, it seems like both sould be optional since they are optional for Prometheus.

Whether to comment them out or set as empty is a bit of a preference but I think it's more clear this way.

**Comment by savanipoojan78** [comment]
Hello @neuronull The Changes you Suggested LGTM.

**Review by spencergilbert — APPROVED** [review]

**Review by neuronull — APPROVED** [review]
LGTM but looks like the chart version needs to be bumped as the CI job is failing

https://github.com/vectordotdev/helm-charts/blob/a15371512cf3cd1110cfff3e6576a211b4dc21b5/charts/vector/Chart.yaml#L3

**Comment by neuronull** [comment]
Ah, it looks like the `README.md` also needs to be updated to include the new options:

```
diff --git a/charts/vector/README.md b/charts/vector/README.md
index e[8](https://github.com/vectordotdev/helm-charts/actions/runs/4401676754/jobs/7718212316#step:4:9)1d6e0..fbea426 100644
--- a/charts/vector/README.md
+++ b/charts/vector/README.md
@@ -1,6 +1,6 @@
 # Vector
 
-![Version: 0.20.1](https://img.shields.io/badge/Version-0.20.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.28.1-distroless-libc](https://img.shields.io/badge/AppVersion-0.28.1--distroless--libc-informational?style=flat-square)
+![Version: 0.20.2](https://img.shields.io/badge/Version-0.20.2-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 0.28.1-distroless-libc](https://img.shields.io/badge/AppVersion-0.28.1--distroless--libc-informational?style=flat-square)
 
 [Vector](https://vector.dev/) is a high-performance, end-to-end observability data pipeline that puts you in control of your observability data. Collect, transform, and route all your logs, metrics, and traces to any vendors you want today and any other vendors you may want tomorrow. Vector enables dramatic cost reduction, novel data enrichment, and data security where you need it, not where is most convenient for your vendors.
 
@@ -17[9](https://github.com/vectordotdev/helm-charts/actions/runs/4401676754/jobs/7718212316#step:4:10),[11](https://github.com/vectordotdev/helm-charts/actions/runs/4401676754/jobs/7718212316#step:4:12) +179,[13](https://github.com/vectordotdev/helm-charts/actions/runs/4401676754/jobs/7718212316#step:4:14) @@ helm install --name <RELEASE_NAME> \
 | podMonitor.enabled | bool | `false` | If true, create a PodMonitor for Vector. |
 | podMonitor.honorLabels | bool | `false` | If true, honor_labels is set to true in the [scrape config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config). |
 | podMonitor.honorTimestamps | bool | `true` | If true, honor_timestamps is set to true in the [scrape config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config). |
+| podMonitor.interval | string | `nil` | Override the interval at which metrics should be scraped. |
 | podMonitor.jobLabel | string | `"app.kubernetes.io/name"` | Override the label to retrieve the job name from. |
 | podMonitor.metricRelabelings | list | `[]` | [MetricRelabelConfigs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#metric_relabel_configs) to apply to samples before ingestion. |
 | podMonitor.path | string | `"/metrics"` | Override the path to scrape. |
 | podMonitor.port | string | `"prom-exporter"` | Override the port to scrape. |
 | podMonitor.relabelings | list | `[]` | [RelabelConfigs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config) to apply to samples before scraping. |
+| podMonitor.scrapeTimeout | string | `nil` | Override the timeout after which the scrape is ended. |
 | podPriorityClassName | string | `""` | Set the [priorityClassName](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) on Vector Pods. |
 | podSecurityContext | object | `{}` | Allows you to overwrite the default [PodSecurityContext](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) for Vector Pods. |
 | psp.create | bool | `false` | If true, create a [PodSecurityPolicy](https://kubernetes.io/docs/concepts/security/pod-security-policy/) resource. PodSecurityPolicy is deprecated as of Kubernetes v1.[21](https://github.com/vectordotdev/helm-charts/actions/runs/4401676754/jobs/7718212316#step:4:22), and will be removed in v1.[25](https://github.com/vectordotdev/helm-charts/actions/runs/4401676754/jobs/7718212316#step:4:26). Intended for use with the "Agent" role. |
Error: Process completed with exit code 1.
```

**Comment by savanipoojan78** [comment]
🫣 Updated the Docs

**Comment by neuronull** [comment]
Thanks for the contribution @savanipoojan78 !

**Pull request merged**